### PR TITLE
feat(dashboard): support bind specific ip(port or ip:port).

### PR DESCRIPTION
**Pull request by zhongwencool**
Fix the: https://github.com/emqx/emqx/pull/5198

Allow to configure listening address in dashboard.

### CODE REVIEW

**Comment by zhongwencool** [comment]
be associated with https://github.com/emqx/minirest/pull/85

File: apps/emqx_dashboard/src/emqx_dashboard.erl [hunk 1]
```diff
@@ -61,15 +61,26 @@ start_listeners() ->
...
+                    end, [], listeners()),
+    case Res of
+        [] -> ok;
+        _ -> {error, Res}
+    end.
 
 stop_listeners() ->
     [begin
-        ok = minirest:stop(Name),
+        _ = minirest:stop(Name),
```
Comment (mononym) [inline apps/emqx_dashboard/src/emqx_dashboard.erl:83]: I assume this means we don't care whether or not there is actually a listener to stop? Would it be worth it to have a case here that logs the fact that there was no listener with 'Name' found?

File: apps/emqx_dashboard/src/emqx_dashboard_schema.erl [hunk 2]
```diff
@@ -25,8 +25,18 @@ namespace() -> <<"dashboard">>.
 roots() -> ["dashboard"].
 
 fields("dashboard") ->
-    [ {listeners, hoconsc:array(hoconsc:union([hoconsc:ref(?MODULE, "http"),
-                                               hoconsc:ref(?MODULE, "https")]))}
+    [ {listeners,
+        sc(hoconsc:array(hoconsc:union([hoconsc:ref(?MODULE, "http"),
+            hoconsc:ref(?MODULE, "https")])),
+            #{ desc =>
+"""HTTP(s) listeners identified by their protocol type,
```
Comment (mononym) [inline apps/emqx_dashboard/src/emqx_dashboard_schema.erl:32]: nit: Add comma after 'listeners' and change 'is' in the next line to 'are' so it reads: HTTP(s) listeners, identified by their protocol type, are used...

Alternatively could rephrase slightly to be: HTTP(s) listeners are identified by their protocol type and are used...

**Review by JimMoen — APPROVED** [review]

File: rebar.config [hunk 3]
```diff
@@ -56,7 +56,7 @@
     , {mria, {git, "https://github.com/emqx/mria", {tag, "0.1.5"}}}
     , {ekka, {git, "https://github.com/emqx/ekka", {tag, "0.11.3"}}}
     , {gen_rpc, {git, "https://github.com/emqx/gen_rpc", {tag, "2.8.0"}}}
-    , {minirest, {git, "https://github.com/emqx/minirest", {tag, "1.2.10"}}}
+    , {minirest, {git, "https://github.com/emqx/minirest", {tag, "1.2.11"}}}
     , {ecpool, {git, "https://github.com/emqx/ecpool", {tag, "0.5.2"}}}
```
[the remark this answers is not in the document]
Comment (zhongwencool) [reply rebar.config:60]: already updated ：）

**Pull request merged**